packages/contracts-bedrock/src/libraries/Errors.sol

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+////////////////////////////////////////////////////////////////
+//                       Common Errors                        //
+////////////////////////////////////////////////////////////////
+
+/// @notice Thrown when an unauthorized address attempts to interact with a contract.
+error BadAuth(string expectedRole);
```

Comment:
`bytes32` is also nice and we just keep role names short enough.

Also a nit: Why `BadAuth` instead of the more common `Unauthorized`?

Reply:
Actually, I agree with enum because it forces a finite number of clearly defined roles that we can match 1:1 with https://docs.optimism.io/chain/security/privileged-roles

We can have overloads that take 2+ args for when there are multiple allowed roles

Reply:
@mds1 reasoning for enum makes sense to me, it will help us clearly define roles

Reply:
If we go with an enum, we could do so by moving this enum from Specs.t.sol into a library: 

https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/test/Specs.t.sol#L31-L42

I would probably want to improve on those names first though.

The scope of this work is expanding, so I am going to move this into draft status until I have a chance to write up a short design doc to get agreement before proceeding.